tests/integration/test_babysit.py

```diff
@@ -0,0 +1,127 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from copr.v3 import Client
+from flexmock import flexmock
+
+from packit.config import PackageConfig, JobConfig, JobType, JobConfigTriggerType
+from packit_service.models import CoprBuildModel, JobTriggerModelType
+from packit_service.service.events import CoprBuildEvent
+from packit_service.worker.build.babysit import check_copr_build
+from packit_service.worker.handlers import CoprBuildEndHandler
+
+
+def test_check_copr_build_no_build():
+    flexmock(CoprBuildModel).should_receive("get_all_by_build_id").with_args(
+        1
+    ).and_return([])
+    assert check_copr_build(build_id=1)
+
+
+def test_check_copr_build_not_ended():
+    flexmock(CoprBuildModel).should_receive("get_all_by_build_id").with_args(
+        1
+    ).and_return([flexmock()])
+    flexmock(Client).should_receive("create_from_config_file").and_return(
+        flexmock(
+            build_proxy=flexmock()
+            .should_receive("get")
+            .with_args(1)
+            .and_return(flexmock(ended_on=False))
+            .mock()
+        )
+    )
+    assert not check_copr_build(build_id=1)
+
+
+def test_check_copr_build_already_successful():
+    flexmock(CoprBuildModel).should_receive("get_all_by_build_id").with_args(
+        1
+    ).and_return([flexmock(status="success")])
+    flexmock(Client).should_receive("create_from_config_file").and_return(
+        flexmock(
+            build_proxy=flexmock()
+            .should_receive("get")
+            .with_args(1)
+            .and_return(flexmock(ended_on="timestamp", state="completed"))
+            .mock()
+        )
+    )
+    assert check_copr_build(build_id=1)
+
+
+def test_check_copr_build_updated():
+    flexmock(CoprBuildModel).should_receive("get_all_by_build_id").with_args(
+        1
+    ).and_return(
+        [
+            flexmock(
+                status="pending",
+                target="the-target",
+                owner="the-owner",
+                project_name="the-project-name",
+                commit_sha="123456",
+                job_trigger=flexmock(type=JobTriggerModelType.pull_request)
+                .should_receive("get_trigger_object")
+                .and_return(
+                    flexmock(
+                        project=flexmock(
+                            repo_name="repo_name",
+                            namespace="the-namespace",
+                            project_url="https://github.com/the-namespace/repo_name",
+                        ),
+                        pr_id=5,
+                        job_config_trigger_type=JobConfigTriggerType.pull_request,
+                    )
+                )
+                .mock(),
+            )
```

Comment:
you sure are adventurous, I didn't even know you can do this with flexmock

Reply:
Yes, but I would not consider this as a nice and clean test-case. (But don't want to spend too much time on it now.)